pyiron/vasp/base.py

```diff
@@ -1341,14 +1341,17 @@ def get_nelect(self):
         else:
             return self["output/generic/dft/n_elect"]
 
-    def get_magnetic_moments(self):
+    def get_magnetic_moments(self, iteration_step=-1):
         """
         Gives the magnetic moments of a calculation for each iteration step.
 
+        Args:
+            iteration_step (int): Step for which the structure is requested
+
         Returns:
             numpy.ndarray: array of final magmetic moments
         """
-        return self['output/generic/dft/final_magmoms']
+        return self['output/generic/dft/final_magmoms'][iteration_step]
```

Comment:
```
magmoms = self['output/generic/dft/final_magmoms']
if magmoms is not None: 
    return magmoms[iteration_step]
else: 
    return None
```

Reply:
is already done 😄 Look at the latest commit.

Reply:
But you read the data twice from the HDF5 file, while I suggest to read it once, store it in a variable and then use this variable twice.

Reply:
That's right, your way is a little bit faster, my way consumes less memory. I don't think that any of them is improving speed or memory consumption.

Reply:
No, file access is always terrible, in particular when using a shared file system on an HPC cluster. While `self['output/generic/dft/final_magmoms']` looks simple, it opens the file, reads the data and closes the file again, that is not worth saving the memory.

Reply:
Fair enough.